src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AwaitExpressionSpiller.cs

```diff
@@ -1111,6 +1044,23 @@ public override BoundNode VisitConditionalReceiver(BoundConditionalReceiver node
             }
         }
 
+        public override BoundNode VisitLambda(BoundLambda node)
+        {
+            var oldCurrentMethod = _F.CurrentMethod;
+            _F.CurrentMethod = node.Symbol;
+            var result = base.VisitLambda(node);
+            _F.CurrentMethod = oldCurrentMethod;
+            return result;
+        }
+
+        public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatement node)
+        {
+            var oldCurrentMethod = _F.CurrentMethod;
+            _F.CurrentMethod = node.Symbol;
```

Comment:
>VisitLambda [](start = 30, length = 11)

We don't need a separate `SpillSequenceSpiller` instance when visiting lamba and local function bodies?

Reply:
Correct, because things are spilled up to the statement level. Lambdas and local functions have a top-level block containing statements at this stage.

---
In reply to: [177557467](https://github.com/dotnet/roslyn/pull/25697#discussion_r177557467) [](ancestors = 177557467)